sys/boot/i386/common/bootargs.h

```diff
@@ -64,10 +64,28 @@ struct bootargs
 	 */
 };
 
+#ifdef LOADER_GELI_SUPPORT
+#include <crypto/intake.h>
+#endif
+
 struct geli_boot_args
 {
     uint32_t		size;
-    char		gelipw[256];
+    union {
+        char            gelipw[256];
+        struct {
+            char                notapw;	/* 
```

Comment:
notapw is the first byte of the keybuf side of this union.
If we have a newer gptzfsboot talking to an older zfsloader (say you switch boot environments to an old one), it will get a null byte as the first byte of the password, and prompt for it instead of trying to use binary data as the key.

We did a union here because we don't want to break the older loaders, which just check the size of the struct to decide if they are new enough to use the geliboot features. Maybe a version number would have been better, as it'll be quite some time before we can remove gelipw[256]